src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -228,7 +245,41 @@ public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, lon
                 }
 
                 if (NetEventSource.Log.IsEnabled()) Trace($"Sending request: {request}");
+                Http3WebtransportSession webtransportSession;
+                if (request.IsWebTransportH3Request)
+                {
+                    webtransportSession = new Http3WebtransportSession(requestStream.quicStream, WTManager!);
+                    WTManager!.AddSession(requestStream.quicStream, webtransportSession);
+                    Task<HttpResponseMessage> responseWebtransportTask = requestStream.SendAsync(cancellationToken);
+                    HttpResponseMessage response = await responseWebtransportTask.ConfigureAwait(false);
+
+                    if (response.IsSuccessStatusCode)
```

Comment:
In the end, I believe Http3Connection is not the place to have all this code. You are mixing up responsibilities.

Could you please think of a way to
1) Have IsSuccessStatusCode check and the following aborts inside **Http3WebtransportSession.ConnectAsync**
2) Have the content set inside Http3RequestStream.SendAsync (also note that you are currently overwriting whatever was initially set in Content)
3) Create a session inside Http3RequestStream.SendAsync when creating the content

Reply:
@CarnaViire Regarding the third one, I cannot create the session when I create the content because I need the session to be created before the request is sent. Otherwise, if the server accepts a session and sends a stream right afterwards, there is a risk of receiving the stream before the session is created. I need to create the session before sending the request. 
I agree with you that the session is best created in the Http3RequestStream file since the session is tied to the request stream, but I think it should be created before the creation of the encoded request (specifically the call of the function `BufferedHeaders`)
What do you think?

Reply:
I updated with 1 and 2 just like you suggested, and 3 as I proposed, If it needs modifications let me know

Reply:
Thanks for the explanation. The idea is ok, I will check the implementation later